### PR TITLE
Remove default proxy; instead use null

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: iothub/device/samples/DeviceReconnectionSample/DeviceReconnectionSample.cs [hunk 1]
```diff
@@ -51,7 +51,7 @@ internal DeviceReconnectionSample(ApplicationParameters parameters)
             if (deviceConnectionStrings == null
                 || !deviceConnectionStrings.Any())
             {
-                throw new ArgumentException("At least one connection string must be provided.", nameof(deviceConnectionStrings));
+                throw new InvalidOperationException("At least one connection string must be provided.");
```
Comment (drwill-ms) [inline iothub/device/samples/DeviceReconnectionSample/DeviceReconnectionSample.cs:54]: Fixed some suggestions in this sample while I was here.

File: iothub/device/src/TransportSettings/IotHubClientAmqpSettings.cs [hunk 2]
```diff
@@ -35,11 +35,6 @@ public sealed class IotHubClientAmqpSettings : IotHubClientTransportSettings
         public IotHubClientAmqpSettings(IotHubClientTransportProtocol transportProtocol = IotHubClientTransportProtocol.Tcp)
         {
             Protocol = transportProtocol;
-            if (Protocol == IotHubClientTransportProtocol.WebSocket)
-            {
-                Proxy = DefaultWebProxySettings.Instance;
```
Comment (drwill-ms) [inline iothub/device/src/TransportSettings/IotHubClientAmqpSettings.cs:40]: Defaults to null, so no need to initialize.

File: iothub/device/src/TransportSettings/IotHubClientTransportSettings.cs [hunk 3]
```diff
@@ -28,29 +27,27 @@ public abstract class IotHubClientTransportSettings
...
+        /// <seealso href="https://docs.microsoft.com/dotnet/api/system.net.http.httpclienthandler.proxy?view=net-6.0"/>
         /// <example>
-        /// To set a proxy you must instantiate an instance of the <see cref="WebProxy"/> class--or any class that derives from
-        /// <see cref="IWebProxy"/>. The snippet below shows a method that returns a device using a proxy that connects to localhost
-        /// on port 8888.
+        /// To set a proxy you must instantiate an instance of the <see cref="WebProxy"/> class--or any class that derives from <see cref="IWebProxy"/>.
+        /// The snippet below shows a method that returns a device using a proxy that connects to localhost on port 8888.
         /// <c>
-        /// static DeviceClient GetClientWithProxy()
+        /// static ServiceClient GetServiceClient()
```
Comment (timtay-microsoft) [inline iothub/device/src/TransportSettings/IotHubClientTransportSettings.cs:38]: Copypaste error, this should be demonstrating device client code

**Review by timtay-microsoft — APPROVED** [review]

File: iothub/device/src/TransportSettings/IotHubClientTransportSettings.cs (hunk 3 above)
Comment (drwill-ms) [reply iothub/device/src/TransportSettings/IotHubClientTransportSettings.cs]: Fixed

**Review by abhipsaMisra — APPROVED** [review]

**Pull request merged**